pangeo_forge_recipes/aggregation.py

```diff
@@ -0,0 +1,154 @@
+from __future__ import annotations
+
+from dataclasses import dataclass, field
+from typing import Dict, Optional
+
+import numpy as np
+
+
+class DatasetCombineError(Exception):
+    pass
+
+
+def empty_xarray_schema():
+    return {"attrs": None, "coords": None, "data_vars": None, "dims": None, "chunks": None}
+
+
+@dataclass
+class XarrayCombineAccumulator:
+    schema: dict = field(default_factory=empty_xarray_schema)
+    concat_dim: Optional[str] = None
+
+    def add_input(self, s: dict, position: int) -> None:
+        s = s.copy()  # avoid modifying input
```

Comment:
Is a deepcopy prudent here?

Reply:
Good question. The only reason I make a copy in the first place is to add the `chunks` key. I need to `copy` for the test suite, where schema fixtures are shared between different invocations of the test. In the wild, I imagine that the input `s` will almost always be discarded after calling `add_input`. The dicts can potentially be very large, so there would be a performance cost to `deepcopy`.

Reply:
That makes sense to me. If you don't depend too much on the values (esp, nested values), then a copy sounds sufficient.